### PR TITLE
Implement dpnp.allclose() for a device without fp64 aspect

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR proposes to provide support of `dpnp.allclose` function for a device without fp64 aspect.
Additionally, use cases with `nan` and `inf` values in input arrays are handling properly now.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp [hunk 1]
```diff
@@ -127,8 +127,79 @@ DPCTLSyclEventRef (*dpnp_all_ext_c)(DPCTLSyclQueueRef,
...
+            {
+                if (std::isinf(array1[i]) || std::isinf(array2[i])) {
+                    partial &= (array1[i] == array2[i]);
+                    continue;
+                }
+            }
+
+            // casting integeral to floating type to avoid bad behavior
+            // on abs(MIN_INT), which leads to undefined result
+            using _Arr2Type = std::conditional_t<std::is_integral_v<_DataType2>,
```
Comment (vlad-perevezentsev) [inline dpnp/backend/kernels/dpnp_krnl_logic.cpp:180]: It would be better if `_Arr2Type` is declared outside the loop to avoid unnecessary computations

Comment (antonwolfy) [reply dpnp/backend/kernels/dpnp_krnl_logic.cpp:180]: but it's done at compile stage, what kind of computation do you mean?

File: dpnp/dpnp_iface.py [hunk 2]
```diff
@@ -453,6 +454,28 @@ def get_usm_ndarray_or_scalar(a):
...
 
+def is_supported_array_or_scalar(a):
+    """
+    Return ``True`` if `a` is a scalar or an array of either
+    :class:`dpnp.ndarray` or :class:`dpctl.tensor.usm_ndarray` type,
+    ``False`` otherwise.
+
+    Parameters
+    ----------
+    a : array
```
Comment (vlad-perevezentsev) [inline dpnp/dpnp_iface.py:465]: `a` may be a scalar

File: dpnp/backend/kernels/dpnp_krnl_logic.cpp (hunk 1 above)
Comment (vlad-perevezentsev) [reply dpnp/backend/kernels/dpnp_krnl_logic.cpp:180]: Yes, you are right.
My remark was more about improving the readability of code but your changes also look good.

**Review by vlad-perevezentsev — APPROVED** [review]
great job!
thank you @antonwolfy

**Pull request merged**